uint8 group_id
uint8 num_groups
string group_name

